extensions/BMO/lib/Reports/WhatsNext.pm

```diff
@@ -0,0 +1,443 @@
+              AND COALESCE(tracking_flags_bugs_2.value, '---') != 'disabled'
+              AND COALESCE(tracking_flags_bugs_3.value, '---') != 'disabled'
+            ORDER BY bugs.delta_ts, bugs.bug_id";
+
+  my $bugs           = get_bug_list($query, $user->id);
+  my $formatted_bugs = format_bug_list($bugs, $user);
+  my $filtered_bugs  = filter_secure_bugs($formatted_bugs);
+
+  return $filtered_bugs;
+}
+
+# Regressions
+sub regression_bugs {
+  my $user = shift;
+  my $dbh  = Bugzilla->dbh;
+
+  # Preselected values for inserting into SQL
+  my $cache      = Bugzilla->process_cache->{whats_next};
+  my $keyword_id = $cache->{regression_id};
+  my $class_ids  = join ',', @{$cache->{classification_ids}};
+  my $bug_states = join ',', map { $dbh->quote($_) } BUG_STATE_OPEN;
+
+  my $query = SELECT . "
+         FROM bugs JOIN products ON bugs.product_id = products.id
+              JOIN keywords ON bugs.bug_id = keywords.bug_id
+        WHERE products.classification_id IN ($class_ids)
+              AND keywords.keywordid = $keyword_id
+              AND bugs.bug_status IN ($bug_states)
+              AND bugs.assigned_to = ?
+     ORDER BY bugs.delta_ts, bugs.bug_id";
+
+  my $bugs           = get_bug_list($query, $user->id);
+  my $formatted_bugs = format_bug_list($bugs, $user);
+  my $filtered_bugs  = filter_secure_bugs($formatted_bugs);
+
+  return $filtered_bugs;
+}
+
+# Other needinfos (needinfos for me but not set by me)
+sub other_needinfo_bugs {
+  my $user = shift;
+  my $dbh  = Bugzilla->dbh;
+
+  # Cached values for inserting into SQL
+  my $cache       = Bugzilla->process_cache->{whats_next};
+  my $needinfo_id = $cache->{needinfo_flag_id};
+  my $class_ids   = join ',', @{$cache->{classification_ids}};
+  my $bug_states  = join ',', map { $dbh->quote($_) } BUG_STATE_OPEN;
+
+  my $query = SELECT . "
+         FROM bugs JOIN products ON bugs.product_id = products.id
+              LEFT JOIN flags AS requestees_login_name ON bugs.bug_id = requestees_login_name.bug_id
+                AND COALESCE(requestees_login_name.requestee_id, 0) = ?
+                AND COALESCE(requestees_login_name.setter_id, 0) != ?
+        WHERE products.classification_id IN ($class_ids)
+              AND bugs.bug_status IN ($bug_states)
+              AND (requestees_login_name.bug_id IS NOT NULL
+                    AND requestees_login_name.type_id = $needinfo_id)
+        ORDER BY bugs.delta_ts, bugs.bug_id";
+
+  my $bugs           = get_bug_list($query, $user->id, $user->id);
+  my $formatted_bugs = format_bug_list($bugs, $user);
+  my $filtered_bugs  = filter_secure_bugs($formatted_bugs);
+
+  return $filtered_bugs;
+}
+
+sub report {
+  my $vars  = shift;
+  my $user  = Bugzilla->user;
+  my $input = Bugzilla->input_params;
+
+  $user->in_group('mozilla-employee-confidential') || ThrowUserError(
+    'auth_failure',
+    {
+      group  => 'mozilla-employee-confidential',
+      action => 'run',
+      object => 'whats_next'
+    }
+  );
```

Comment:
Slight miscommunication here, when I said the report should only be visible to employees, I meant just the icon in the header.  There's no need for the dashboard itself to be restricted.

Reply:
Will fix.